ui/src/app/pages/data/cohort/cohort-resource-card.tsx

```diff
@@ -31,89 +32,87 @@ export const CohortResourceCard = fp.flow(
   constructor(props: Props) {
     super(props);
     this.state = {
-      showRenameModal: false
+      showRenameModal: false,
+      referencingDataSets: [],
     };
   }
 
-  get resourceType(): ResourceType {
-    return ResourceType.COHORT;
-  }
-
-  get displayName(): string {
-    return this.props.resource.cohort.name;
-  }
-
-  // this is duplicated
-  get canWrite(): boolean {
-    return this.props.resource.permission === 'OWNER'
-      || this.props.resource.permission === 'WRITER';
-  }
-
-  get canDelete(): boolean {
-    return this.props.resource.permission === 'OWNER';
-  }
-
-  get resourceUrl(): string {
-    const {workspaceNamespace, workspaceFirecloudName, cohort} = this.props.resource;
-    return `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}` +
-      `/data/cohorts/build?cohortId=${cohort.id}`;
-  }
-
-  get reviewCohortUrl(): string {
+  get reviewUrlForCohort(): string {
     const {workspaceNamespace, workspaceFirecloudName, cohort} = this.props.resource;
 
     return `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}` +
       `/data/cohorts/${cohort.id}/review`;
   }
 
   get actions(): Action[] {
+    const {resource} = this.props;
     return [
       {
         icon: 'note',
         displayName: 'Rename',
         onClick: () => {
           this.setState({showRenameModal: true});
         },
-        disabled: !this.canWrite
+        disabled: !canWrite(resource)
       },
       {
         icon: 'copy',
         displayName: 'Duplicate',
         onClick: () => this.duplicate(),
-        disabled: !this.canWrite
+        disabled: !canWrite(resource)
       },
       {
         icon: 'pencil',
         displayName: 'Edit',
-        onClick: () => navigateByUrl(this.resourceUrl),
-        disabled: !this.canWrite
+        onClick: () => navigateByUrl(getResourceUrl(resource)),
+        disabled: !canWrite(resource)
       },
       {
         icon: 'grid-view',
         displayName: 'Review',
-        onClick: () => navigateByUrl(this.reviewCohortUrl),
-        disabled: !this.canWrite
+        onClick: () => navigateByUrl(this.reviewUrlForCohort),
+        disabled: !canWrite(resource)
       },
       {
         icon: 'trash',
         displayName: 'Delete',
         onClick: () => {
-          this.props.showConfirmDeleteModal(this.displayName,
-            this.resourceType, () => this.delete());
+          this.props.showConfirmDeleteModal(getDisplayName(resource),
+            getType(resource), () => this.delete());
         },
-        disabled: !this.canDelete
+        disabled: !canDelete(resource)
       }
     ];
   }
 
-  delete() {
-    return cohortsApi().deleteCohort(
-      this.props.resource.workspaceNamespace,
-      this.props.resource.workspaceFirecloudName,
-      this.props.resource.cohort.id)
-      .then(() => {
-        this.props.onUpdate();
-      });
+  async setOrClearDataSetList() {
```

Comment:
The logic in this extracted and renamed code was hard to follow.  I may try to simplify it.